### ISSUE TITLE
candidate.cfg is not sufficiently unique if napalm-aos is used in parallel

### ISSUE THREAD

**Issue by cs-1** [napalm-automation-community/napalm-aos#13]
I use napalm-aos in napalm-ansible. Normally, Ansible will run processes in parallel. However, napalm-aos always uses "candidate.cfg" as default candidate configuration in https://github.com/napalm-automation-community/napalm-aos/blob/1fdb92ee0ee35b272c9a4c0999ca1f2873f10ac3/napalm_aos/aos.py#L92 This name is used e.g. in `load_merge_candidate()`. When `load_merge_candidate` is called by two processes running in parallel, using the same "candidate.cfg" name will break the `load_merge_candidate` process.

I suggest making the default "candidate.cfg" more unique. You already create sufficiently unique filenames in https://github.com/napalm-automation-community/napalm-aos/blob/1fdb92ee0ee35b272c9a4c0999ca1f2873f10ac3/napalm_aos/aos.py#L234 however, since you set "candidate.cfg" in https://github.com/napalm-automation-community/napalm-aos/blob/1fdb92ee0ee35b272c9a4c0999ca1f2873f10ac3/napalm_aos/aos.py#L92 that's the one that is used. One patch would be to replace that line
```
self.candidate_cfg_file = optional_args.get('candidate_cfg_file', 'candidate.cfg')
```
with
```
self.candidate_cfg_file = optional_args.get('candidate_cfg_file', str(uuid.uuid4()))
```

**Comment by cs-1** [comment]
The matter seems to be more complicated. You use `self.candidate_cfg_file` for different purposes, if I'm correct. You use it as a temporary filename on the host that runs napalm and you also use it as a filename when copying the candidate configuration to the switch (which is never removed -- I'll open another issue for that). I don't know if the above patch breaks the double use of the filename.

**Comment by cs-1** [comment]
I've proposed a patch in pull request https://github.com/napalm-automation-community/napalm-aos/pull/20 to fix this problem, however, we can only test the code in limited scenarios so it need thorough testing.

**Comment by napalmaos** [comment]
I just have merged your pull request #20 to pull request #22. Please help to test.
Thanks.

**Comment by napalmaos** [comment]
I use candidate_remote_cfg_file to reformat load_replace_candidate and load_merge_candidate in  pull request #23. 
I think we should not use helper function, it will make a mismatch between ReplaceConfigException and MergeConfigException.
Please help to test again.
Thanks

**Comment by vmuthukrishnan** [comment]
Hi, Kindly close the issue if it is resolved, otherwise let us know if there any issues still present.
